Fixes some hostile mob targeting

emmanuelbassil opened the pull request:
🆑 emmanuelbassil
bugfix: Antlions now properly burrow and appear near/on you to ruin your day.
bugfix: King goats have re-learned the lost skill of explorer bowling, and have a chance of running you over.
bugfix: Charbabys now have a 10% chance per attack of setting you on fire again.
/🆑 

Also fixed leeches becoming hungrier if they're pursuing you; fixed vagrants responding to whoever shoots them.
But the hunger feature for leeches isn't really used much (might expand in the future), and vagrants being the OP little weasels that they are aren't in game.
Replaced antlion's attack() by resolve_attackby since I know its specific weapon has no attack() override. Didn't do it for the others simple_mobs as I'm unsure, it's done in the refactor however.

MuckerMayhem commented on code/game/atoms_health.dm:
```diff
@@ -88,8 +88,6 @@
 	SHOULD_CALL_PARENT(TRUE)
 	if (!get_max_health())
 		return FALSE
-	if (health_dead())
-		return FALSE
```
I'm a little worried what the side effects of removing this will be. What is fixed by removing this?

emmanuelbassil replied:
Answered in discord.

Spookerton commented on code/game/atoms_health.dm:
```diff
@@ -88,8 +88,6 @@
 	SHOULD_CALL_PARENT(TRUE)
 	if (!get_max_health())
 		return FALSE
-	if (health_dead())
-		return FALSE
```
link to the answer given:
<https://discord.com/channels/104289531796131840/256853514624434177/1140450361455427765>


Do this for /mob/living instead.

emmanuelbassil replied:
Fixed.

SierraKomodo commented on code/game/atoms_health.dm:
```diff
@@ -101,6 +101,19 @@
 		return FALSE
 	. = ..()
 
+/mob/living/can_damage_health(damage, damage_type)
+	SHOULD_CALL_PARENT(FALSE)
+	if (status_flags & GODMODE)
+		return FALSE
+	if (!get_max_health())
+		return FALSE
+	if (!damage || damage < health_min_damage)
+		return FALSE
+	if (is_damage_immune(damage_type))
+		return FALSE
+	return TRUE
+
```
It's safe to simply remove the `health_dead()` check from the parent proc. This way we're not copy+pasting a bunch of checks.

SierraKomodo commented on code/game/atoms_health.dm:
```diff
@@ -101,6 +101,19 @@
 		return FALSE
 	. = ..()
 
+/mob/living/can_damage_health(damage, damage_type)
+	SHOULD_CALL_PARENT(FALSE)
+	if (status_flags & GODMODE)
+		return FALSE
+	if (!get_max_health())
+		return FALSE
+	if (!damage || damage < health_min_damage)
+		return FALSE
+	if (is_damage_immune(damage_type))
+		return FALSE
+	return TRUE
+
```
It's safe to simply remove the `health_dead()` check from the parent procs, to avoid copy+pasting everything but that line.

emmanuelbassil replied:
Fixed

emmanuelbassil committed:
Fixes some mob targeting


Auto stash before rebase of "refs/heads/AssortedFixes"
Auto stash before rebase of "refs/heads/eyestab"
